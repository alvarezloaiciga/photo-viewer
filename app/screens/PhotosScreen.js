import React, { Component } from 'react';
import { StyleSheet, ScrollView, View, Text, TouchableHighlight, Image, Dimensions } from 'react-native';
import { SearchBar } from 'react-native-elements';

import PhotosGrid from '../components/PhotosGrid';
import PhotosApi from '../services/PhotosApi';

const PHOTO_API_CONSUMER_KEY = "9t4nuIALhDuQCbwLRKXOofBJ2kRIC0YECWsEMuHD"
const PHOTO_API_SEARCH_URL = `https://api.500px.com/v1/photos/search?consumer_key=${PHOTO_API_CONSUMER_KEY}&image_size=600`;

export default class PhotosScreen extends Component {
  static navigationOptions = {
    title: 'Photos',
  };

  constructor(props) {
    super(props)
    this.state = {
      photos: []
    }

    this.searchPhotos = this.searchPhotos.bind(this)
    this.viewPhoto = this.viewPhoto.bind(this)
  };

  searchPhotos(term) {
    const that = this

    PhotosApi
      .searchPhotos(term)
      .then((responseJson) => {
        that.setState({
          photos: responseJson.photos
        })
      })
      .catch((error) => {
        console.error(error)
      });
  };

  viewPhoto(photo) {
    const { navigate }  = this.props.navigation
    navigate('PhotoDetail', { photo: photo })
  };

  render() {
    let photos = <Text style={styles.noPhotos}>No Photos</Text>

    if (this.state.photos.length > 0) {
      photos = <PhotosGrid photos={this.state.photos} onClickedPhoto={this.viewPhoto}/>
    }

    return(
      <ScrollView style={styles.scrollView}>
        <SearchBar
          round
          lightTheme
          onChangeText={this.searchPhotos}
          placeholder='Search Photos'/>

        <View style={styles.containerView}>
          { photos }
        </View>
      </ScrollView>
    )
  }
}


const styles = StyleSheet.create({
  scrollView: {},
  containerView:  {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap'
  },

  noPhotos: {
  }
})
