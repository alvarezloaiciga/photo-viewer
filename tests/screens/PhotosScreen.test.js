import 'react-native';
import React from 'react';
import PhotosScreen from '../../app/screens/PhotosScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <PhotosScreen />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
