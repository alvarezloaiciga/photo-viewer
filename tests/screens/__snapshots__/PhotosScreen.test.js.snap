// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={Object {}}
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#393e42",
            "borderBottomColor": "#000",
            "borderBottomWidth": 1,
            "borderTopColor": "#000",
            "borderTopWidth": 1,
          },
          Object {
            "backgroundColor": "#e1e8ee",
            "borderBottomColor": "#e1e1e1",
            "borderTopColor": "#e1e1e1",
          },
          undefined,
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search Photos"
        placeholderTextColor="#86939e"
        selectionColor="#86939e"
        style={
          Array [
            Object {
              "backgroundColor": "#303337",
              "borderRadius": 3,
              "color": "#86939e",
              "fontSize": 17.5,
              "height": 30,
              "margin": 8,
              "overflow": "hidden",
              "paddingLeft": 26,
              "paddingRight": 19,
            },
            Object {
              "backgroundColor": "#bdc6cf",
            },
            false,
            Object {
              "borderRadius": 15,
            },
            undefined,
            undefined,
            false,
          ]
        }
        underlineColorAndroid="transparent"
      />
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 16,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "position": "absolute",
                "top": 15.5,
              },
              Object {
                "left": 16,
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î¢¶
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={Object {}}
      >
        No Photos
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
